src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private int size;
+    private Object[] elementData;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_SIZE];
+
+    }
+
+    public void checkCapacity() {
+        if (size == elementData.length) {
+            int newSize = (Integer) elementData.length * 3 / 2;
+            Object[] newArray = new Object[newSize];
+            System.arraycopy(elementData, 0, newArray, 0, size);
+            elementData = newArray;
+        }
+    }
+
     @Override
     public void add(T value) {
+        checkCapacity();
+        elementData[size++] = value;
 
     }
 
     @Override
     public void add(T value, int index) {
```

Comment:
The method `add(T value, int index)` should allow adding an element at the end of the list. The current check `checkIndexForSet(index)` does not allow adding at `size` index, which is valid for adding at the end.